docs/DeveloperGuide.md

```diff
@@ -1,11 +1,216 @@
+---
```

Comment:
Maybe the use of short code snippets in the DG would be good for users to improve their understanding.